Api/Api.cs

```diff
@@ -613,5 +615,35 @@ public virtual void Dispose()
         {
             // NOP
         }
+
+
+        /// <summary>
+        /// Generate a secure hash for the authorization headers.
+        /// </summary>
+        /// <returns>Time based hash of user token and timestamp.</returns>
+        public static string CreateSecureHash(int timestamp, string token)
```

Comment:
Arguably, this need not be public. But if it is public, perhaps a better name is `CreateTimestampedToken` or even `CreatePassword`